src/protocols/secure_channel/CASESession.cpp

```diff
@@ -328,7 +298,7 @@ CHIP_ERROR CASESession::DeriveSecureSession(SecureSession & session, SecureSessi
         // TODO: Add IPK to Salt
         bbuf.Put(mMessageDigest, sizeof(mMessageDigest));
 
-        VerifyOrReturnError(bbuf.Fit(), CHIP_ERROR_NO_MEMORY);
+        VerifyOrReturnError(bbuf.Fit(), CHIP_ERROR_BUFFER_TOO_SMALL);
     }
 
     ReturnErrorOnFailure(session.InitFromSecret(ByteSpan(mSharedSecret, mSharedSecret.Length()), ByteSpan(msg_salt.Get(), saltlen),
```

Comment:
IPK used needs to be accounted-for in salt for spec compliance. See:

![image](https://user-images.githubusercontent.com/1313847/126553668-04ce5df4-e9f3-4bbe-93a4-6afcdcfd3472.png)

Also, I'm concerned that since digest is finished() in SigmaR3, that receiving more messages out of order may mess-up order. Strongly recommend finishing digest in here just above the `bbuf.Put()`.

Reply:
Updated